frontend/src/Components/ProfilePage/ProfilePage.jsx

```diff
@@ -0,0 +1,231 @@
+import React, { useState, useEffect } from "react";
+import Modal from "../Modal/Modal";
+import SideBar from "../SideBar/SideBar";
+import { Tab, Tabs, TabList, TabPanel } from "react-tabs";
+import "react-tabs/style/react-tabs.css";
+import "./ProfilePage.css";
+
+const ProfilePage = () => {
+  const [profile, setProfile] = useState({
+    name: "",
+    email: "",
+    location: "",
+    bio: "",
+    interests: "",
+    profilePicture: "",
+  });
+  const [selectedFile, setSelectedFile] = useState(null);
+  const [pinnedServices, setPinnedServices] = useState([]);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchProfile = async () => {
+      const token = localStorage.getItem("token");
+
+      try {
+        const response = await fetch("http://localhost:3000/api/user/profile", {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${token}`,
+          },
+        });
+
+        if (response.ok) {
+          const data = await response.json();
+          setProfile(data);
+          setPinnedServices(data.pinnedServices || []);
+        } else {
+          console.error("Failed to fetch profile data");
+        }
+      } catch (error) {
+        console.error("Error fetching profile data:", error);
+      }
+    };
+
+    fetchProfile();
+  }, []);
+
+  const handleChange = (e) => {
+    setProfile({ ...profile, [e.target.name]: e.target.value });
+  };
+
+  const handleFileChange = (e) => {
+    setSelectedFile(e.target.files[0]);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const token = localStorage.getItem("token");
+
+    try {
+      const formData = new FormData();
+      formData.append("name", profile.name);
+      formData.append("email", profile.email);
+      formData.append("location", profile.location);
+      formData.append("bio", profile.bio);
+      formData.append("interests", profile.interests);
+      if (selectedFile) {
+        formData.append("profilePicture", selectedFile);
+      }
+
+      const response = await fetch("http://localhost:3000/api/user/profile", {
+        method: "PUT",
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+        body: formData,
+      });
+
+      if (response.ok) {
+        const updatedProfile = await response.json();
+        setProfile(updatedProfile);
+        setIsModalOpen(false); // Close the modal on successful update
+        alert("Profile updated successfully");
+      } else {
+        console.error("Failed to update profile");
+        alert("Failed to update profile");
+      }
+    } catch (error) {
+      console.error("Error updating profile:", error);
+      alert("Failed to update profile");
+    }
+  };
+
+  return (
+    <div className="profile-page">
+      <button
```

Comment:
this button doesn't seem to have an accessible label, since its descendant is an icon.  I think we'd want to add an aria-label to this button for Screen Readers.

Reply:
Done